src/pages/settings/Security/AddDelegate/ValidateCodeForm/BaseValidateCodeForm.tsx

```diff
@@ -0,0 +1,208 @@
+    wrapperStyle?: StyleProp<ViewStyle>;
+};
+
+function BaseValidateCodeForm({autoComplete = 'one-time-code', innerRef = () => {}, delegate, role, wrapperStyle}: BaseValidateCodeFormProps) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const [formError, setFormError] = useState<ValidateCodeFormError>({});
+    const [validateCode, setValidateCode] = useState('');
+    const inputValidateCodeRef = useRef<MagicCodeInputHandle>(null);
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const login = account?.primaryLogin;
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing -- nullish coalescing doesn't achieve the same result in this case
+    const focusTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+    const currentDelegate = account?.delegatedAccess?.delegates?.find((d) => d.email === delegate);
+    const validateLoginError = ErrorUtils.getLatestErrorField(currentDelegate, 'addDelegate');
+
+    const shouldDisableResendValidateCode = !!isOffline || currentDelegate?.isLoading;
+
+    useImperativeHandle(innerRef, () => ({
+        focus() {
+            inputValidateCodeRef.current?.focus();
+        },
+        focusLastSelected() {
+            if (!inputValidateCodeRef.current) {
+                return;
+            }
+            if (focusTimeoutRef.current) {
+                clearTimeout(focusTimeoutRef.current);
+            }
+            focusTimeoutRef.current = setTimeout(() => {
+                inputValidateCodeRef.current?.focusLastSelected();
+            }, CONST.ANIMATED_TRANSITION);
+        },
+    }));
+
+    useFocusEffect(
+        useCallback(() => {
+            if (!inputValidateCodeRef.current) {
+                return;
+            }
+            if (focusTimeoutRef.current) {
+                clearTimeout(focusTimeoutRef.current);
+            }
+            focusTimeoutRef.current = setTimeout(() => {
+                inputValidateCodeRef.current?.focusLastSelected();
+            }, CONST.ANIMATED_TRANSITION);
+            return () => {
+                if (!focusTimeoutRef.current) {
+                    return;
+                }
+                clearTimeout(focusTimeoutRef.current);
+            };
+        }, []),
+    );
+
+    /**
+     * Request a validate code / magic code be sent to verify this contact method
+     */
+    const resendValidateCode = () => {
+        if (!login) {
+            return;
+        }
+        Delegate.requestValidationCode();
+
+        inputValidateCodeRef.current?.clear();
+    };
+
+    /**
+     * Handle text input and clear formError upon text change
+     */
+    const onTextInput = useCallback(
+        (text: string) => {
+            setValidateCode(text);
+            setFormError({});
+            if (validateLoginError) {
+                Delegate.clearDelegatorErrors();
```

Comment:
@fedirjh when we will dimiss the error, the modal will close because of the useEffect in `DelegateMagicCodePage` 

Do you have any ideas on how we can fix this?

Reply:
im thinking we should add a pending field: `validateCode`

Reply:
how about we add a pending field `validateCodePending`

it'll clear only for successData

Reply:
that is a nice idea

Reply:
Updated existing delegate from array to take care of dupes

https://github.com/Expensify/App/pull/47353#issuecomment-2334596009